### PR TITLE
Python plugin jinja template

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

Addresses **Approach 1(A)**, but using a python measure: https://github.com/NREL/OpenStudio/issues/4634#issuecomment-1346858424

I've created two measures:
* `PythonPluginRuby` (uses `python_plugin_program_erb.py` template)
* `PythonPluginPython` (uses `python_plugin_program_jinja.py` template)

Both measures are called by `compact_python_plugin.osw`.

```
$ /c/openstudio-3.6.0-alpha/bin/openstudio.exe labs run -w resources/Examples/compact_osw/compact_python_plugin.osw
```

Appears to be working successfully.

Update: I've removed `sys.path.append` and tried running with OS v3.7.0 (w/out labs), and it appears to be working successfully.

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: resources/Examples/compact_osw/files/python_plugin_program_erb.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from pyenergyplus.plugin import EnergyPlusPlugin
+
+import os, sys
```
Comment (jmarrec) [inline resources/Examples/compact_osw/files/python_plugin_program_erb.py:3]:
```suggestion
import os
import sys
```

File: resources/Examples/compact_osw/files/python_plugin_program_erb.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+sys.path.append('C:/Python38/Lib/site-packages') # this should (needs to?) be same version as E+'s version
```
Comment (jmarrec) [inline resources/Examples/compact_osw/files/python_plugin_program_erb.py:4]: We're going to have to make this path to system python's site_packages some kind of cmake replacement or a command line arg if we want to test this.

But more broadly, this is going to pose the question of whether we should/need to provide some common python packages out of the box like we do for ruby gems (and take into account the pain that it is to maintain that stuff). I'm specifically thinking about native packages (related to this issue https://github.com/NREL/OpenStudio/issues/4846), (pandas does have a C extension as well, I think something about numpy may be special)

@kbenne @tijcolem

Comment (jmarrec) [reply resources/Examples/compact_osw/files/python_plugin_program_erb.py:5]: Following https://github.com/NREL/OpenStudio/pull/4868 you can make your test pass the `--python_home` or `--python_path` and it'll work. 

I think we should discuss whether we should always add some common python packages like Jinja2 (https://github.com/NREL/OpenStudio/issues/4906#issuecomment-1621966528)

Comment (joseph-robertson) [reply resources/Examples/compact_osw/files/python_plugin_program_erb.py:5]: @jmarrec @kbenne 

Has there been any more discussion/movement on #4906?

Comment (jmarrec) [reply resources/Examples/compact_osw/files/python_plugin_program_erb.py:5]: See here: https://github.com/NREL/OpenStudio/blob/0ab02805c62a5f459e669bf3f4c65c95a1d8eed7/src/cli/CMakeLists.txt#L343-L393

**Comment by joseph-robertson** [comment]
@jmarrec This appears to be working now (i.e., can import pandas and jinja2). How can we formalize running `compact_python_plugin.osw` as a test? Add something to src/cli/CMakeLists?

File: resources/Examples/compact_osw/measures/PythonPluginPython/measure.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import os, sys
```
Comment (jmarrec) [inline resources/Examples/compact_osw/measures/PythonPluginPython/measure.py:1]:
```suggestion
import os
import sys
```

Generally speaking you should `pip install black isort` on your system and reformat your python files via

```shell
isort .
black -l 120 .
```

File: resources/Examples/compact_osw/compact_python_plugin.osw
Comment (jmarrec) [inline resources/Examples/compact_osw/compact_python_plugin.osw:1]: Move this to another folder. Or it'll get installed. Suggestion for the layout of `resources/Example`

```diff
 .
 ├── compact_osw
 ├── with_analysis
+└── python_plugin
```

File: resources/Examples/compact_osw/files/python_plugin_program_erb.py
Comment (jmarrec) [inline resources/Examples/compact_osw/files/python_plugin_program_erb.py:1]: This should be inside the measures directory inside resources/ , so it is a proper measure and can be moved around.

**Comment by jmarrec** [comment]
> @jmarrec This appears to be working now (i.e., can import pandas and jinja2). How can we formalize running `compact_python_plugin.osw` as a test? Add something to src/cli/CMakeLists?

Yes. like you've done. But it bothers me we don't check at all if E+ actually runs the python plugins

**Comment by jmarrec** [comment]
@joseph-robertson I've made some adjustments (moved things around, linted) + actually tested the plugins are running.

Assuming CI comes clean and you're ok with the changes, this can drop.

**Pull request merged**